### PR TITLE
Enable hiprtc by default

### PR DESCRIPTION
**Pull request by pfultz2**
It will still be disabled when using `BUILD_DEV=On`.

### CODE REVIEW

**Review by TedThemistokleous — APPROVED** [review]

**Review by umangyadav — APPROVED** [review]

**Comment by causten** [comment]
Maybe we should hold off until you review why all the tests are failing

**Comment by umangyadav** [comment]
Performance tests are failing because, it requires `MIGRAPHX_ENABLE_HIPRTC_WORKAROUNDS=1` flag.

File: src/targets/gpu/CMakeLists.txt [hunk 1]
```diff
@@ -33,7 +33,11 @@ if(NOT TARGET MIOpen)
     message(SEND_ERROR "Cant find miopen")
 endif()
 
-set(MIGRAPHX_USE_HIPRTC OFF CACHE BOOL "Use hipRTC APIs")
+if(BUILD_DEV)
+    set(MIGRAPHX_USE_HIPRTC OFF CACHE BOOL "Use hipRTC APIs")
+else()
+    set(MIGRAPHX_USE_HIPRTC ON CACHE BOOL "Use hipRTC APIs")
```
Comment (umangyadav) [inline src/targets/gpu/CMakeLists.txt:39]: Can we  use `rocm_find_program_version` on `amdclang++` ? 

If it is less than 16.0.0  we can `set(ENV{MIGRAPHX_ENABLE_HIPRTC_WORKAROUNDS} True)`

Comment (pfultz2) [reply src/targets/gpu/CMakeLists.txt:39]: That only sets it for cmake time and not runtime. Also, there is performance impact with the workarounds so we don't want to enable it by default.

**Comment by pfultz2** [comment]
> Performance tests are failing because, it requires MIGRAPHX_ENABLE_HIPRTC_WORKAROUNDS=1 flag.

Performance tests should build with `-DMIGRAPHX_USE_HIPRTC=Off` since the workarounds do have performance impact.

**Comment by umangyadav** [comment]
> > Performance tests are failing because, it requires MIGRAPHX_ENABLE_HIPRTC_WORKAROUNDS=1 flag.
> 
> Performance tests should build with `-DMIGRAPHX_USE_HIPRTC=Off` since the workarounds do have performance impact.

They would fail to compile with hipRTC then.

**Comment by umangyadav** [comment]
> > They would fail to compile with hipRTC then.
> 
> Thats why they should disable it.

You mean we should be using `clang++` as compiler for performance tests ?

**Comment by pfultz2** [comment]
> You mean we should be using clang++ as compiler for performance tests ?

Yes at least until they upgrade to a working hiprtc.

**Comment by umangyadav** [comment]
@causten this one is ready to go. Performance tests are working.

**Pull request merged**